kroxylicious/src/main/java/io/kroxylicious/proxy/internal/codec/DecodePredicate.java

```diff
@@ -21,11 +24,13 @@
  */
 public interface DecodePredicate {
     static DecodePredicate forFilters(KrpcFilter... filters) {
+
+        List<FilterInvoker> invokers = Arrays.stream(filters).map(FilterInvokers::from).toList();
```

Comment:
Are we missing a trick here to turn the array of filters into a compound invoker here? 

It doesn't really make any difference to the performance profile but I think it gives better encapsulation.

Reply:
Similar to the other thing, it's tricky to make a composite because we don't have access to `apiVersion` in `onRequest/onResponse`. You can make the composite return true from the `shouldHandle[Request|Response]` methods if any filter matches the key/version, but then you can't locate the same invoker again in `on[Request|Response]` because we don't have apiVersion at hand.

Reply:
maybe instead we could move `DecodePredicate` to filter-api and rename it `HandlePredicate` and have `Invoker` implement it? Then we could make a composite `HandlePredicate` without worrying about how onRequest/Response compose.

Reply:
@robobario and I discussed this on a call. We came to the conclusion there wasn't enough value in `HandlePredicate` to justify it. However @robobario is planning to add `apiVersion` to the `onRequest|onResponse` signatures in a separate PR as they make it easier to generate custom messages in the proxy and are probably necessary for correctness in certain situations.